Compiler/src/abstractinterpretation.jl

```diff
@@ -3079,6 +3079,13 @@ function abstract_eval_call(interp::AbstractInterpreter, e::Expr, sstate::Statem
     end
 end
 
+function is_field_pointerfree(dt::DataType, fidx::Int)
+    dt.layout == C_NULL && return false
```

Comment:
I would add a type-assert to avoid invalidations, see https://github.com/JuliaLang/julia/pull/59886:
```suggestion
    dt.layout::Ptr{Cvoid} == C_NULL && return false
```